File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3SearchResult.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3SearchResult {
    /// <summary>
    /// Train stations, tram stops, bus stops, regional coach stops or Night Bus stops
    /// </summary>
    /// <value>Train stations, tram stops, bus stops, regional coach stops or Night Bus stops</value>
    [DataMember(Name="stops", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "stops")]
    public List<V3ResultStop> Stops { get; set; }

    /// <summary>
    /// Train lines, tram routes, bus routes, regional coach routes, Night Bus routes
    /// </summary>
    /// <value>Train lines, tram routes, bus routes, regional coach routes, Night Bus routes</value>
    [DataMember(Name="routes", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "routes")]
    public List<V3ResultRoute> Routes { get; set; }

    /// <summary>
    /// myki ticket outlets
    /// </summary>
    /// <value>myki ticket outlets</value>
    [DataMember(Name="outlets", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "outlets")]
    public List<V3ResultOutlet> Outlets { get; set; }

    /// <summary>
    /// Gets or Sets Status
    /// </summary>
    [DataMember(Name="status", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "status")]
    public V3Status Status { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3SearchResult {\n");
      sb.Append("  Stops: ").Append(Stops).Append("\n");
      sb.Append("  Routes: ").Append(Routes).Append("\n");
      sb.Append("  Outlets: ").Append(Outlets).Append("\n");
      sb.Append("  Status: ").Append(Status).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: java-vertx/src/main/java/io/swagger/server/api/model/V3SearchResult.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.server.api.model.V3ResultOutlet;
import io.swagger.server.api.model.V3ResultRoute;
import io.swagger.server.api.model.V3ResultStop;
import io.swagger.server.api.model.V3Status;
import java.util.ArrayList;
import java.util.List;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3SearchResult
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3SearchResult   {
  
  private List<V3ResultStop> stops = null;

  
  private List<V3ResultRoute> routes = null;

  
  private List<V3ResultOutlet> outlets = null;

  private V3Status status = null;

  public V3SearchResult stops(List<V3ResultStop> stops) {
    this.stops = stops;
    return this;
  }

  public V3SearchResult addStopsItem(V3ResultStop stopsItem) {
    if (this.stops == null) {
      this.stops = new ArrayList<V3ResultStop>();
    }
    this.stops.add(stopsItem);
    return this;
  }

  /**
   * Train stations, tram stops, bus stops, regional coach stops or Night Bus stops
   * @return stops
   **/
    public List<V3ResultStop> getStops() {
    return stops;
  }

  public void setStops(List<V3ResultStop> stops) {
    this.stops = stops;
  }

  public V3SearchResult routes(List<V3ResultRoute> routes) {
    this.routes = routes;
    return this;
  }

  public V3SearchResult addRoutesItem(V3ResultRoute routesItem) {
    if (this.routes == null) {
      this.routes = new ArrayList<V3ResultRoute>();
    }
    this.routes.add(routesItem);
    return this;
  }

  /**
   * Train lines, tram routes, bus routes, regional coach routes, Night Bus routes
   * @return routes
   **/
    public List<V3ResultRoute> getRoutes() {
    return routes;
  }

  public void setRoutes(List<V3ResultRoute> routes) {
    this.routes = routes;
  }

  public V3SearchResult outlets(List<V3ResultOutlet> outlets) {
    this.outlets = outlets;
    return this;
  }

  public V3SearchResult addOutletsItem(V3ResultOutlet outletsItem) {
    if (this.outlets == null) {
      this.outlets = new ArrayList<V3ResultOutlet>();
    }
    this.outlets.add(outletsItem);
    return this;
  }

  /**
   * myki ticket outlets
   * @return outlets
   **/
    public List<V3ResultOutlet> getOutlets() {
    return outlets;
  }

  public void setOutlets(List<V3ResultOutlet> outlets) {
    this.outlets = outlets;
  }

  public V3SearchResult status(V3Status status) {
    this.status = status;
    return this;
  }

  /**
   * Get status
   * @return status
   **/
    public V3Status getStatus() {
    return status;
  }

  public void setStatus(V3Status status) {
    this.status = status;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3SearchResult v3SearchResult = (V3SearchResult) o;
    return Objects.equals(this.stops, v3SearchResult.stops) &&
        Objects.equals(this.routes, v3SearchResult.routes) &&
        Objects.equals(this.outlets, v3SearchResult.outlets) &&
        Objects.equals(this.status, v3SearchResult.status);
  }

  @Override
  public int hashCode() {
    return Objects.hash(stops, routes, outlets, status);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3SearchResult {\n");
    
    sb.append("    stops: ").append(toIndentedString(stops)).append("\n");
    sb.append("    routes: ").append(toIndentedString(routes)).append("\n");
    sb.append("    outlets: ").append(toIndentedString(outlets)).append("\n");
    sb.append("    status: ").append(toIndentedString(status)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
